File: cs/Refactor1/SRP_DI_Workshop/OrderProcessor.cs
using SRP_DI_Workshop.Domain.OrderOperations;
using SRP_DI_Workshop.ServiceInterface;

namespace SRP_DI_Workshop
{
    public class OrderProcessor
    {
        private readonly Deduplicator _deduplicator = new Deduplicator();
        private readonly InventoryRepository _InventoryRepository = new InventoryRepository();
        private readonly OrderRepository _orderRepository = new OrderRepository();
        private readonly LoggerService _loggerService = new LoggerService();
        private readonly ResponseMessageFactory _responseMessageFactory = new ResponseMessageFactory();

        public ResponseMessage ProcessOrder(RequestMessage reqMsg)
        {
            ResponseMessage responseToDuplicateMessage = _deduplicator.GetExistingResponseMessage(reqMsg.RequestId);

            if (responseToDuplicateMessage != null)
                return responseToDuplicateMessage;

            ResponseMessage response;

            switch (reqMsg.Operation)
            {
                case CreateOrderOperation.Id:
                    CreateOrderOperation submitOrder = new CreateOrderOperation
                    {
                        InventoryRepository = _InventoryRepository,
                        OrderRepository = _orderRepository
                    };

                    response = submitOrder.ExecuteOperation(reqMsg);
                    break;
                case CancelOrderOperation.Id:
                    CancelOrderOperation cancelOrder = new CancelOrderOperation
                    {
                        InventoryRepository = _InventoryRepository,
                        OrderRepository = _orderRepository
                    };

                    response = cancelOrder.ExecuteOperation(reqMsg);
                    break;
                case GetOrderOperation.Id:
                    GetOrderOperation getOrder = new GetOrderOperation
                    {
                        OrderRepository = _orderRepository
                    };

                    response = getOrder.ExecuteOperation(reqMsg);
                    break;
                default:
                    _loggerService.WriteLine("Received bad message: " + reqMsg.RequestId, "ProcessOrderError");
                    response = _responseMessageFactory.CreateBadRequestResponseMessage(reqMsg);
                    break;
            }

            _deduplicator.StoreResponseToMessage(reqMsg.RequestId, response);

            return response;
        }
    }
}

File: java/Refactor1/src/main/java/no/bouvet/solid/srpdip/OrderProcessor.java
package no.bouvet.solid.srpdip;

import no.bouvet.solid.srpdip.domain.orderoperations.CancelOrderOperation;
import no.bouvet.solid.srpdip.domain.orderoperations.CreateOrderOperation;
import no.bouvet.solid.srpdip.domain.orderoperations.GetOrderOperation;
import no.bouvet.solid.srpdip.messageinterface.RequestMessage;
import no.bouvet.solid.srpdip.messageinterface.ResponseMessage;

public class OrderProcessor {

	private Deduplicator deduplicator = new Deduplicator();
	private InventoryRepository inventoryRepository = new InventoryRepository();
	private OrderRepository orderRepository = new OrderRepository();

	public ResponseMessage processMessage(RequestMessage reqMsg) {
		ResponseMessage responseToDuplicateMessage = deduplicator.getExistingResponseMessage(reqMsg.getRequestId());

		if (responseToDuplicateMessage != null) {
			return responseToDuplicateMessage;
		}

		ResponseMessage response = null;

		switch (reqMsg.getOperation()) {
		case SUBMIT_ORDER:
			response = new CreateOrderOperation(inventoryRepository, orderRepository).execute(reqMsg);
			break;
		case CANCEL_ORDER:
			response = new CancelOrderOperation(inventoryRepository, orderRepository).execute(reqMsg);
			break;
		case GET_ORDER_DETAILS:
			response = new GetOrderOperation(orderRepository).execute(reqMsg);
		}

		deduplicator.storeResponseToMessage(reqMsg.getRequestId(), response);

		return response;
	}
}
